src/Moryx.Notifications/Adapter/NotificationAdapter.cs

```diff
@@ -222,17 +228,25 @@ void INotificationSourceAdapter.PublishProcessed(INotification notification)
                 _pendingPubs.Remove(map);
                 _published.Add(map);
                 _listLock.ExitWriteLock();
+                return;
             }
-            else
-            {
-                // Notification is maybe not pending anymore - we only can acknowledge it
-                _listLock.ExitWriteLock();
 
-                var managed = (IManagedNotification)notification;
+            // Notification is maybe not pending anymore
+            _listLock.ExitWriteLock();
+            var managed = (IManagedNotification)notification;
+
+            // If necessary we can acknowledge it
+            if (managed.Acknowledged is null)
+            {
+                Logger?.Log(LogLevel.Error, "Notification was removed from the pending publications " +
+                    "before being published but is not acknowledged.");
                 managed.Acknowledged = DateTime.Now;
                 managed.Acknowledger = nameof(NotificationAdapter);
                 Acknowledged?.Invoke(this, notification);
             }
+
+            Logger?.Log(LogLevel.Warning, $"Notification was removed from the pending publications. " +
+                $"It was already acknowledged by {0} at {1}.", managed.Acknowledger, managed.Acknowledged);
```

Comment:
This would be `It was already acknowledged by 0 at 1` - you have to remove the `$`